### ISSUE TITLE
vkAllocateMemory not handling null pointers

### ISSUE THREAD

**Issue by yavn** [KhronosGroup/Vulkan-LoaderAndValidationLayers#321]
When memory allocation fails, e.g. due to `VK_ERROR_OUT_OF_HOST_MEMORY` from a custom allocator, layer keeps using the returned NULL pointer. This leads to the following error message:

> MEM(ERROR): object: 0 type: 8 location: 4663 msgCode: 11: Mem Object 0 has not been freed. You should clean up this memory by calling vkFreeMemory(0) prior to vkDestroyDevice().

I think the problem is in `vkAllocateMemory` function in `core_validation.cpp` which calls

```
add_mem_obj_info(my_data, device, *pMemory, pAllocateInfo);  // pMemory is NULL
```

This problem is demonstrated by the test `dEQP-VK.api.object_management.alloc_callback_fail.device_memory_small` available in CTS development branch.

**Comment by mark-lunarg** [comment]
Addressed by PR #1150 .

**Comment by yavn** [comment]
Confirmed fixed, thank you.
